FEATURE: Custom Scripts Arguments Overhaul (#101)
Dungeons-of-Kathallion/Bane-Of-Wargs: docs/ENGINE_FUNCTIONS.md

| encounter_text_show()   | `player`, `item`, `enemy`, `map`, `map_location`, `enemies_remaining`, `lists`, `defeat_percentage`, `preferences`, `drinks`, `npcs`, `zone`, `mounts`, `mission`, `start_player`, `dialog`, `text_replacements_generic`, `player_damage_coefficient`, `previous_player`, `save_file`, `start_time`, `enemies_damage_coefficient`, `entry_data`, `enemies`, `no_run_away`=False | Displays the encountering text of enemy pool list `entry_data`                                          |
| fight()                 | `player`, `item`, `enemy`, `map`, `map_location`, `enemies_remaining`, `lists`, `preferences`, `drinks`, `npcs`, `start_player`, `zone`, `dialog`, `mission`, `mounts`, `player_damage_coefficient`, `start_time`, `text_replacements_generic`,  `previous_player`, `save_file`, `enemies_damage_coefficient`, `defeat_percentage`, `entry_data`, `enemies`                   | Starts fight of enemy pool list `entry_data`                                                            |

## check_yaml.py

> The [`check_yaml.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/check_yaml.py) class won't be a class you'll use for custom scripting, as it contains the functions to run tests on the loaded game data. But if you're interested in contributing, you might want to check yourself the class and analyze it.

Reviewer: ```suggestion
The [`check_yaml.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/check_yaml.py) class won't be a class you'll use for custom scripting, as it contains the functions to run tests on the loaded game data. But if you're interested in contributing, you might want to check out the class yourself and analyze it.
```

## colors.py

> The [`colors.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/colors.py) class is a database of color codes, used for colored terminal input. Just check the class if you need a color code you're searching for.

Reviewer: ```suggestion
The [`colors.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/colors.py) class is a database of color codes; used for colored terminal input. Just check the class if you need a color code you're searching for.
```

|----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|
| print_dialog() | `current_dialog`, `dialog`, `preferences`, `text_replacements_generic`, `player`, `drinks`, `item`, `enemy`, `npcs`, `start_player`, `lists`, `zone`, `mission`, `mounts`, `start_time`, `map`, `save_file`, `player_damage_coefficient`, `enemies_damage_coefficient`, `previous_player`, `mission_offered`=None | Triggers dialog `current_dialog`. Specify a mission id at `mission_offered` if you want conversation actions like 'accept()' to be ran |

## dungeon.py

> The [`dungeon.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/dungeon.py) class handles the dungeon UI. It only contains one function, that starts the dungeon loop:

Reviewer: ```suggestion
The [`dungeon.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/dungeon.py) class handles the dungeon UI. It only contains one function, one that starts the dungeon loop:
```

## logger_sys.py

The [`logger_sys.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/logger_sys.py) class handles the logging system of the program:

| Name            | Arguments | Description                                                                                                  |
|-----------------|-----------|--------------------------------------------------------------------------------------------------------------|
| write_message() | `message` | Writes string `message` to the game logs                                                                      |
| log_message()   | `message` | Processes string `message` to the write it to the program's logs (run level tests with the player preferences) |

## main.py

> The [`main.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/main.py) class doesn't have any function, as it is the game engine's root, that contains the master loop of the game.

Reviewer: ```suggestion
The [`main.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/main.py) class doesn't have any function, as it is the game engine's root containing the master loop of the game.
```

## npc_handling.py

The [`npc_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/npc_handling.py) class handles the interaction between the player and the npcs:

| Name       | Arguments                                                                | Description                                                                                                      |
|------------|--------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
| init_npc() | `map_location`, `player`, `npcs`, `drinks`, `item`, `preferences`, `map` |  Inits the interaction loop between the player and the npc at the player's current map point digit `map_location` |

## player_handling.py

> The [`player_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/player_handling.py) class handles all major actions that can be done on the player, as death:

Reviewer: ```suggestion
The [`player_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/player_handling.py) class handles all major actions that can be done on the player, such as death:
```
```suggestion
The [`player_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/player_handling.py) class handles all major actions that can be done on the player, as in death:
```
Not sure which because there is only one function documented here.

| cout()                    | `text`="", `end`="\n"    | Writes to stdout text `text` with at the end `end`, and then flush it to stdout                                                                |
| cinput()                  | `text`=""                | Returns gotten user input, with an optional text `text` before it                                                                              |
| cinput_multi()            | `text`="", `type`="list" | Returns gotten user input through multiple line, until a EOFE break (Ctrl + D), and return                                                     |
| show_menu()               | `options`, `length`=52   | Displays a choice box with list of choices `options`. The `length` argument is deprecated since [v0.23-alpha](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/releases/tag/0.23-alpha), as it's automatically determined. Returns the selected choice as a string |
| format_string_separator() | `text`                   | Formats `text` to add thousands separators to every number present in it, and returns the formatted string                                     |

## text_handling.py

> The [`text_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/text_handling.py) class contains many useful functions to format and/or output text, that are often used:

Reviewer: ```suggestion
The [`text_handling.py`](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/blob/master/source/text_handling.py) class contains many useful functions to format and/or output text that are often used:
```
